modules/ivc_champva/app/services/ivc_champva/pdf_stamper.rb

```diff
@@ -78,25 +144,62 @@ def self.stamp(desired_stamp, stamped_template_path, append_to_stamp: false, tex
+        Rails.logger.info 'IVC Champva Forms - PdfStamper: stamp calling verified_multistamp'
         verified_multistamp(stamped_template_path, text, page_configuration, font_size)
       else
-        datestamp_instance = PDFUtilities::DatestampPdf.new(current_file_path, append_to_stamp:)
-        current_file_path = datestamp_instance.run(text:, x:, y:, text_only:, size: 9)
-        File.rename(current_file_path, stamped_template_path)
+        begin
+          Rails.logger.info 'IVC Champva Forms - PdfStamper: stamp creating datestamp instance'
+          datestamp_instance = PDFUtilities::DatestampPdf.new(current_file_path, append_to_stamp:)
+          current_file_path = datestamp_instance.run(text:, x:, y:, text_only:, size: 9)
+          Rails.logger.info 'IVC Champva Forms - PdfStamper: stamp renaming to stamped_template_path'
+          File.rename(current_file_path, stamped_template_path)
+        rescue
+          begin
+            Rails.logger.info 'IVC Champva Forms - PdfStamper: stamp an error occurred, deleting temp file'
+            Common::FileHelpers.delete_file_if_exists(current_file_path)
+          rescue => e
+            Rails.logger.error 'IVC Champva Forms - PdfStamper: stamp error in rescue, logging and swallowing'
+            Rails.logger.error 'IVC Champva Forms - PdfStamper: Failed to clean up temporary file', message: e.message
+            Rails.logger.error e.backtrace.join("\n")
+            # Don't re-raise an error here, the original error is more important
+          end
+          Rails.logger.info 'IVC Champva Forms - PdfStamper: stamp re-raising error'
+          raise
+        end
       end
     end
+    # rubocop:enable Metrics/MethodLength
 
     def self.perform_multistamp(stamped_template_path, stamp_path)
+      Rails.logger.info 'IVC Champva Forms - PdfStamper: entered perform_multistamp'
       out_path = "#{Common::FileHelpers.random_file_path}.pdf"
       pdftk = PdfFill::Filler::PDF_FORMS
+      Rails.logger.info 'IVC Champva Forms - PdfStamper: perform_multistamp calling pdftk.multistamp'
       pdftk.multistamp(stamped_template_path, stamp_path, out_path)
+      Rails.logger.info 'IVC Champva Forms - PdfStamper: perform_multistamp post pdftk.multistamp delete'
       File.delete(stamped_template_path)
+      Rails.logger.info 'IVC Champva Forms - PdfStamper: perform_multistamp post pdftk.multistamp rename'
       File.rename(out_path, stamped_template_path)
     rescue
-      Common::FileHelpers.delete_file_if_exists(out_path)
+      begin
+        Rails.logger.info 'IVC Champva Forms - PdfStamper: perform_multistamp an error occurred, deleting temp file'
+        Common::FileHelpers.delete_file_if_exists(out_path)
+      rescue => e
+        Rails.logger.error 'IVC Champva Forms - PdfStamper: perform_multistamp error in rescue, logging and swallowing'
+        Rails.logger.error 'IVC Champva Forms - PdfStamper: Failed to clean up temporary file', message: e.message
```

Comment:
Noticed you were using sanitized messages above. Just verifying that nothing in here needs to be sanitized.